layers/eight_mile/utils.py

```diff
@@ -1548,3 +1548,16 @@ def mime_type(file_name: str) -> str:
     if is_text_file(b):
         return "text/plain"
     return "application/w2v"
+
+
+@export
+def to_numpy(x):
+    if isinstance(x, np.ndarray):
+        return x
+    try:
+        import torch
```

Comment:
can actually test if its an instance of a torch.Tensor right?

Reply:
Why do we need to test that? If you are using tf1 then your predict result is already a numpy array and is caught by the first check, if you are using tf2 and you don't have pytorch installed the import causes an error which we catch. We then call `.numpy()` to turn the `tf.Tesor` into a numpy array. If you have pytorch installed but it was a tf2 tensor the `.detach()` throws which we again catch and just us `.numpy()`. If we are using pytorch we use `.detach()` which is defined on Tensors with and without `requires_grad=True` and then we move it to `cpu` before calling `.numpy()`

I guess an explicit check (while keeping the try in case they don't have pytorch installed) is probably simpler to reason when we need to look at this function again

Reply:
it could be a little more generalizable in the future, and more precise about what its trying to accomplish.